curriculum/challenges/english/25-front-end-development/workshop-fruit-search-app/67ed4c13de85e0aa05df5145.md

```diff
@@ -0,0 +1,113 @@
+---
+id: 67ed4c13de85e0aa05df5145
+title: Step 7
+challengeType: 0
+dashedName: step-7
+---
+
+# --description--
+
+Your input element should have a `value` attribute set to the `query` state.
+
+# --hints--
+
+You should set the `value` attribute of your input to `query`.
+
+```js
+assert.match(code, /value=\{query\}/)
+```
+
+# --seed--
+
+## --seed-contents--
+
+```html
+<!DOCTYPE html>
+<html>
+<head>
+  <meta charset="UTF-8" />
+  <title>Fruits Search</title>
+   <script src="https://cdnjs.cloudflare.com/ajax/libs/react/18.3.1/umd/react.development.min.js"></script>
+   <script src="https://cdnjs.cloudflare.com/ajax/libs/react-dom/18.3.1/umd/react-dom.development.min.js"></script>
+   <script src="https://cdnjs.cloudflare.com/ajax/libs/babel-standalone/7.26.5/babel.min.js"></script>
+  <script 
+    data-plugins="transform-modules-umd"
+    type="text/babel"
+    src="index.jsx"
+  ></script>
+  <link rel="stylesheet" href="styles.css" />
+</head>
+<body>
+  <div id="root"></div>
+  <script
+    data-plugins="transform-modules-umd"
+    type="text/babel"
+    data-presets="react"
+    data-type="module"
+  >
+    import { FruitsSearch } from './index.jsx';
+    ReactDOM.createRoot(document.getElementById('root')).render(<FruitsSearch />);
+  </script>
+</body>
+</html>
+```
+
+```css
+body {
+  font-family: Arial, sans-serif;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+  background-color: #f4f4f4;
+}
+
+#search-container {
+  text-align: center;
+  background: white;
+  padding: 20px;
+  border-radius: 10px;
+  box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
+}
+
+#search-input {
+  padding: 10px;
+  width: 80%;
+  border: 1px solid #ccc;
+  border-radius: 5px;
+  margin-bottom: 10px;
+}
+
+#results {
+  text-align: left;
+  max-height: 150px;
+  overflow-y: auto;
+}
+.result-item {
+  padding: 5px;
+  border-bottom: 1px solid #ddd;
+}
+```
+
+```jsx
+const { useState, useEffect } = React;
+
+--fcc-editable-region--
+export function FruitsSearch() {
+  const [query, setQuery] = useState('');
+  const [results, setResults] = useState([]);
+
+  return (
+    <div id="search-container">
+      <form>
+        <label htmlFor="search-input">Search for fruits:</label>
+        <input
+          id="search-input"
+          type="text"
+        />
+      </form>
+    </div>
+  );
+--fcc-editable-region--
```

Comment:
we don't need to show the full component in the editable region. just the part that needs to change

```suggestion
export function FruitsSearch() {
  const [query, setQuery] = useState('');
  const [results, setResults] = useState([]);

  return (
    <div id="search-container">
      <form>
        <label htmlFor="search-input">Search for fruits:</label>
        --fcc-editable-region--
        <input
          id="search-input"
          type="text"
        />
        --fcc-editable-region--
      </form>
    </div>
  );
```